netmaster/master/netmaster.go

```diff
@@ -159,11 +162,105 @@ func CreateGlobal(stateDriver core.StateDriver, gc *intent.ConfigGlobal) error {
 		masterGc.FwdMode = gc.FwdMode
 	}
 
+	if len(gcfgUpdateList) > 0 {
+		// Delete old state
+
+		gOper := &gstate.Oper{}
+		gOper.StateDriver = stateDriver
+		err = gOper.Read("")
+		if err == nil {
+			for _, res := range gcfgUpdateList {
+				err = gCfg.UpdateResources(res)
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+
+			for _, res := range gcfgUpdateList {
+				// setup resources
+				err = gCfg.Process(res)
+				if err != nil {
+					log.Errorf("Error updating the config %+v. Error: %s", gCfg, err)
+					return err
+				}
+			}
+		}
+
+		err = gCfg.Write()
+		if err != nil {
+			log.Errorf("error updating global config.Error: %s", err)
+			return err
+		}
+	}
 	err = masterGc.Write()
 	if err != nil {
 		return err
 	}
 
+	return nil
+}
+
+// UpdateGlobal updates the global state
+func UpdateGlobal(stateDriver core.StateDriver, gc *intent.ConfigGlobal) error {
+	log.Infof("Received global update with intent {%v}", gc)
+	var err error
+	gcfgUpdateList := []string{}
+
+	masterGc := &mastercfg.GlobConfig{}
+	masterGc.StateDriver = stateDriver
+	masterGc.Read("global")
+
+	gstate.GlobalMutex.Lock()
+	defer gstate.GlobalMutex.Unlock()
+
+	gCfg := &gstate.Cfg{}
+	gCfg.StateDriver = stateDriver
+	gCfg.Read("global")
+
+	_, vlansInUse := gCfg.GetVlansInUse()
+	_, vxlansInUse := gCfg.GetVxlansInUse()
+
+	// check for valid values
+	if gc.NwInfraType != "" {
+		switch gc.NwInfraType {
+		case "default", "aci", "aci-opflex":
+			// These values are acceptable.
+		default:
+			return errors.New("Invalid fabric mode")
+		}
+		masterGc.NwInfraType = gc.NwInfraType
+	}
```

Comment:
sure we can do that. But we are anyways exiting if invalid? But I can make the change